Cowles.js

```diff
@@ -0,0 +1,316 @@
+		item.date = doc.evaluate("//strong[contains(., 'CFP Date')]", doc, null, XPathResult.ANY_TYPE, null).iterateNext().nextSibling.textContent;
+		item.volume = doc.evaluate("//strong[contains(., 'CFP Vol(Issue)')]", doc, null, XPathResult.ANY_TYPE, null).iterateNext().nextSibling.textContent.split('(')[0];
+		item.issue = doc.evaluate("//strong[contains(., 'CFP Vol(Issue)')]", doc, null, XPathResult.ANY_TYPE, null).iterateNext().nextSibling.textContent.split('(')[1][0];
+		item.pages = doc.evaluate("//strong[contains(., 'CFP page numbers')]", doc, null, XPathResult.ANY_TYPE, null).iterateNext().nextSibling.textContent;
+		author = doc.querySelectorAll("div.comma span.comma a");
+		for (let auth of author) item.creators.push(ZU.cleanAuthor(auth.textContent,"author",false));
+		item.url = doc.evaluate("//strong[contains(., 'CFP Paper Title')]", doc, null, XPathResult.ANY_TYPE, null).iterateNext().nextSibling.getAttribute('href');
+		item.libraryCatalog = "Cowles Foundation";
+		try {
+			item.extra = "See also: " + doc.evaluate("//strong[contains(., 'See CFDP')]", doc, null, XPathResult.ANY_TYPE, null).iterateNext().nextSibling.textContent;
+		}
+		catch (err) {}
+	}
+	else if (url.includes('/cfdp/')) {
+		item = new Zotero.Item("report");
+		item.title = doc.querySelector('h3 a').textContent;
+		item.reportType = "Cowles Foundation Discussion Paper";
+		item.reportNumber = doc.querySelector("#page-title").textContent.match(/\d+/)[0];
+		author = doc.querySelectorAll("div.comma span.comma a");
+		for (let auth of author) item.creators.push(ZU.cleanAuthor(auth.textContent,"author",false));
+
+		item.libraryCatalog = "Cowles Foundation";
+		item.date = doc.evaluate("//strong[contains(., 'Publication Date')]", doc, null, XPathResult.ANY_TYPE, null).iterateNext().nextSibling.textContent;
+		item.pages = doc.evaluate("//strong[contains(., 'Pages')]", doc, null, XPathResult.ANY_TYPE, null).iterateNext().nextSibling.textContent;
+		try {
+			item.abstractNote = doc.evaluate("//strong[contains(., 'Abstract')]", doc, null, XPathResult.ANY_TYPE, null).iterateNext().parentElement.nextSibling.textContent;
+		}
+		catch (err) {}
+		pdfurl = doc.querySelector('h3 a').getAttribute('href');
+		item.attachments.push({
+			title: item.title,
+			mimeType:"application/pdf",
+			url:pdfurl
+		});
+	}
+	else if (url.includes('/cfm-')) {
+		item = new Zotero.Item("book");
+		item.title = doc.querySelector('a[href*="/pub/"]').textContent;
+		item.series = "Cowles Monograph";
+		item.seriesNumber = doc.querySelector("#page-title").textContent.match(/\d+/)[0];
+		item.libraryCatalog = "Cowles Foundation";
+		
+		var status = "";
+		var author = doc.querySelector(".field-name-field-paper-title p a")
+		.previousSibling
+		.textContent
+		.trim()
+		.split(",");
+		if (author[1] == " ed." || author[1] == " eds.") status = "editor";
```

Comment:
there are never more than two authors? Meyer, Smith, and Jones, eds. would break this. I'd prefer something more robust. Something like `if (author.length>1){if (author.slice(-1)[0] == " ed. ")` etc. might work

Reply:
There are, indeed, never more than two authors and no monographs has been published since 1981 (https://cowles.yale.edu/publications/archives/cfm)